hal/armv7m/imxrt/117x/imxrt.c

```diff
@@ -389,6 +415,481 @@ __attribute__((section(".noxip"))) int _imxrt_setLevelLPCG(int clock, int level)
 }
 
 
+void _imxrt_delay(u32 ticks)
+{
+	/* TODO: use better method e.g. count cycles, must not use IRQs and peripheral timers! */
+	while (ticks-- != 0u) {
+		__asm volatile("nop");
+	}
+}
+
+
+void _imxrt_setPllBypass(u8 clk_pll, u8 enable)
+{
+	/* clang-format off */
+	switch (clk_pll) {
+		case clk_pllarm:
+			*(imxrt_common.anadig_pll + arm_pll_ctrl) = (enable != 0) ?
+				(*(imxrt_common.anadig_pll + arm_pll_ctrl) | (1uL << 17u)) :
+				(*(imxrt_common.anadig_pll + arm_pll_ctrl) & ~(1uL << 17u));
+			break;
+		case clk_pllsys1:
+			*(imxrt_common.anadig_pll + sys_pll1_ctrl) = (enable != 0) ?
+				(*(imxrt_common.anadig_pll + sys_pll1_ctrl) | (1uL << 16u)) :
+				(*(imxrt_common.anadig_pll + sys_pll1_ctrl) & ~(1uL << 16u));
+			break;
+		case clk_pllsys2:
+			*(imxrt_common.anadig_pll + sys_pll2_ctrl) = (enable != 0) ?
+				(*(imxrt_common.anadig_pll + sys_pll2_ctrl) | (1uL << 16u)) :
+				(*(imxrt_common.anadig_pll + sys_pll2_ctrl) & ~(1uL << 16u));
+			break;
+		case clk_pllsys3:
+			*(imxrt_common.anadig_pll + sys_pll3_ctrl) = (enable != 0) ?
+				(*(imxrt_common.anadig_pll + sys_pll3_ctrl) | (1uL << 16u)) :
+				(*(imxrt_common.anadig_pll + sys_pll3_ctrl) & ~(1uL << 16u));
+			break;
+		case clk_pllaudio:
+			/* TODO: access through ANATOP AI */
+			break;
+		case clk_pllvideo:
+			/* TODO: access through ANATOP AI */
+			break;
+		default:
+			break;
+	}
+	/* clang-format on */
+}
+
+
+void _imxrt_pmuBypassAnaLdo(void)
+{
+	/* HP mode */
+	*(imxrt_common.anadig_pll + pmu_ldo_lpsr_ana) &= ~1u;
+	_imxrt_delay(1000u * 1000u);
+
+	/* Tracking mode */
+	*(imxrt_common.anadig_pll + pmu_ldo_lpsr_ana) |= 1uL << 19u;
```

Comment:
`1u` vs `1uL` usage is inconsistent, imho better to stick with plain `1u` (as we have `unsigned int` in mind, rather than `unsigned long`)

Reply:
It is because of rule MISRA 20212 12.2 (required).

Reply:
Ok, I understand, we need more than 16 bits, so int might be too small